### PR TITLE
planner: split casetest for pushdown

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #44940

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/casetest/pushdown/push_down_test.go [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+	"github.com/pingcap/tidb/testkit"
+	"github.com/pingcap/tidb/testkit/testdata"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPushLimitDownIndexLookUpReader(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+
+	tk.MustExec("set tidb_cost_model_version=2")
```
Comment (Defined2014) [inline planner/core/casetest/pushdown/push_down_test.go:31]: Maybe it's no need anymore.

File: planner/core/casetest/BUILD.bazel [hunk 2]
```diff
@@ -15,7 +15,6 @@ go_test(
     ],
     data = glob(["testdata/**"]),
     flaky = True,
-    race = "on",
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply planner/core/casetest/BUILD.bazel:18]: 1、 This test cases is too many. so it takes a long time to complete.
2、 We will add the race to this subpackage.

[the remark this answers is not in the document]
Comment (Defined2014) [reply planner/core/casetest/BUILD.bazel:18]: How about add it in pushdown/BUILD.bazel

[the remark this answers is not in the document]
Comment (hawkingrei) [reply planner/core/casetest/BUILD.bazel:18]: added

**Review by Defined2014 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**